tools/train_classifier.py

```diff
@@ -62,11 +63,41 @@ def clean_pipe_state_dict_key(key):
     path_annotations_valid = args['--path_annotations_valid'] or None
     num_layers_to_finetune = int(args['--num_layers_to_finetune'])
     temporal_training = args['--temporal_training']
+    save_checkpoints = args['--save_checkpoints']
+    frequency = int(args['--frequency'] or 5)
+    weights = args['--weights']
+    resume = args['--resume']
+
+    # Create a folder "checkpoints" to save model training weights
+    if save_checkpoints:
+        path = os.path.join(path_in, "checkpoints")
+        os.makedirs(path, exist_ok=True)
 
     # Load feature extractor
     feature_extractor = feature_extractors.StridedInflatedEfficientNet()
-    checkpoint = torch.load('resources/backbone/strided_inflated_efficientnet.ckpt')
-    feature_extractor.load_state_dict(checkpoint)
+    # explanation of resume_epoch is in training_loops function
+    resume_epoch = -1
+    if weights:
+        # Load custom and original classifiers
+        checkpoint_classifier = load_weights(weights)
+        checkpoint = torch.load('resources/backbone/strided_inflated_efficientnet.ckpt')
+        # Update original weights in case some intermediate layers have been finetuned
+        name_finetuned_layers = set(checkpoint.keys()).intersection(checkpoint_classifier.keys())
+        for key in name_finetuned_layers:
+            checkpoint[key] = checkpoint_classifier.pop(key)
+        feature_extractor.load_state_dict(checkpoint['model_state_dict'])
+    elif resume:
+        checkpoint_classifier = load_weights(os.path.join(path_out, 'checkpoints/', 'last_classifier.checkpoint'))
+        checkpoint = torch.load('resources/backbone/strided_inflated_efficientnet.ckpt')
+        name_finetuned_layers = set(checkpoint.keys()).intersection(checkpoint_classifier.keys())
+        for key in name_finetuned_layers:
+            checkpoint[key] = checkpoint_classifier.pop(key)
+        resume_epoch = checkpoint['epoch']
```

Comment:
`checkpoint` here is the loaded backbone weights, not the saved checkpoint file from training, hence the `KeyError`